staging/src/k8s.io/kubectl/pkg/cmd/cp/cp.go

```diff
@@ -283,6 +285,32 @@ func (o *CopyOptions) checkDestinationIsDir(dest fileSpec) error {
 	return o.execute(options)
 }
 
+// checkContainerCommandExist receives a destination fileSpec and
+// determines if the command which kubectl need is on the
+// pod. If the command does not exist , an error is returned.
+func (o *CopyOptions) checkContainerCommandExist(command string, dest fileSpec) error {
+	outBuf := bytes.NewBuffer(make([]byte, 0, 512))
+	options := &exec.ExecOptions{
+		StreamOptions: exec.StreamOptions{
+			IOStreams: genericclioptions.IOStreams{
+				Out:    outBuf,
+				ErrOut: bytes.NewBuffer([]byte{}),
+			},
+			Namespace: dest.PodNamespace,
+			PodName:   dest.PodName,
+		},
+
+		Command:  []string{command},
+		Executor: &exec.DefaultRemoteExecutor{},
+	}
+
+	_ = o.execute(options)
+	if strings.Contains(outBuf.String(), fmt.Sprintf("\"%s\": executable file not found in $PATH", command)) {
```

Comment:
I'm concerned about portability with this error string matching approach. Can we check the error code instead? As @apelisse [pointed out](https://github.com/kubernetes/kubernetes/pull/79075#issuecomment-505659234) on a previous attempt at the same fix: 

> error code 127 usually means that the command was not found: https://www.tldp.org/LDP/abs/html/exitcodes.html

It might be good to cover code 126 too (found but not executable)

Reply:
I did it with checking error code `126`  at first. but I'm not sure if it returns the same code every time。 I thought it should be `127` ，but got `126` every time。

If you ensure that checking `126`,`127` is enougth,I will do that 

BTW，I will do it like below
```go
import utilexec "k8s.io/client-go/util/exec"


	err := o.execute(options)
	if err !=nil {
		if ce, ok:=err.(utilexec.CodeExitError);ok && ce.Code == 126{
			return  fmt.Errorf("command '%s' which kubectl need can not be found in the chosen container", command)
		}
	}
```